src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SslOverTdsStream.cs

```diff
@@ -59,7 +104,14 @@ public override int Read(byte[] buffer, int offset, int count)
                     // Account for split packets
                     while (readBytes < TdsEnums.HEADER_LEN)
                     {
-                        readBytes += _stream.Read(packetData, readBytes, TdsEnums.HEADER_LEN - readBytes);
+                        if (async)
+                        {
+                            readBytes += await _stream.ReadAsync(packetData, readBytes, TdsEnums.HEADER_LEN - readBytes, token).ConfigureAwait(false);
```

Comment:
This library targets netstandard2.0 rather than netcoreapp2.1, right?  If it targeted the latter, it'd be nice to use the new Memory-based overloads, but I don't think it does.

Reply:
Yes, SqlClient targets netstandard2.0. I would like to change it to target netcoreapp 
I will open an issue to track the work. Then SqlClient can benefit more from the Memory Based overloads.